File: 2_Basic_Programming/divide-cal.cs
using System;

namespace divide_cal
{
    class Program
    {
        static void DivideCalculation()
        {

            try
            {
                int dividend;
                int divisor;
                int result;

                Console.Write("Input dividend: ");
                int.TryParse(Console.ReadLine(), out dividend);

                Console.Write("Input divisor: ");
                int.TryParse(Console.ReadLine(), out divisor);

                //Divide Calculation
                result = dividend / divisor;
                Console.WriteLine("Result = {0}", result);
            }

            catch(Exception e)
            {
                Console.WriteLine(e.Message);
            }

            finally
            {
                Console.WriteLine("End Program");
            }
            
        }
        static void Main(string[] args)
        {
            DivideCalculation();
        }
    }
}

File: 2_Basic_Programming/DivideCal.cpp
//MISSION: DIVIDE CALCULATION
//By: PALM

#include <iostream>
using namespace std;

//Calculate X divide by Y
double DivideCal(double x, double y){
	try{

		//Check if y = 0 then throw y
		if (y==0){
			throw y;
		}
		return x/y;
	}
	catch(double y){
		cout << "Exception : y must not be 0.\n";
		throw; //do not want to print result
	}
}

int main(){
	
	// Input X and Y
	double x, y;
	cout << "x : ";
	cin >> x;
	cout << "y : ";
	cin >> y;
		
	cout << "Result = " << DivideCal(x,y);
}